### PR TITLE
Document parallel gmsh

### LINKED ISSUE #609: Parallelizing of Gmsh meshes not clearly documented
I created a 3D spherical mesh in Gmsh 3.0.5 and trying to import to fipy 3.1.3. The following code works when run as `python test_gmsh.py` but crashes when run with `mpirun -np 8 python test_gmsh.py`
```python
#!/usr/bin/env python2
import fipy

mesh = fipy.Gmsh3D('sphere.msh')
```
The error is:
```
Traceback (most recent call last):
  File "test_gmsh.py", line 3, in <module>
    mesh = fipy.Gmsh3D('sphere.msh')
  File "/usr/lib/python2.7/site-packages/fipy/meshes/gmshMesh.py", line 1944, in __init__
    self._orderedCellVertexIDs_data) = self.mshFile.read()
  File "/usr/lib/python2.7/site-packages/fipy/meshes/gmshMesh.py", line 809, in read
    raise GmshException(errStr)
fipy.meshes.gmshMesh.GmshException: Gmsh hasn't produced any cells! Check your Gmsh code.

Gmsh output:
```

If I replace `.msh` file with `.geo` file it works in both cases, however the mesh seems to be different from one generated in Gmsh.

### PR DESCRIPTION
**Pull request by guyer**
* Make clearer how Gmsh meshes are partitioned.
* Document and relax constraints when installing gmsh with conda (constraints were because some versions didn't partition the way FiPy needed, hence including this here).

Fixes #609 
Addresses #697

### CODE REVIEW

**Comment by guyer** [comment]
Failures are unrelated to this PR

File: INSTALLATION.rst [hunk 1]
```diff
@@ -85,7 +88,7 @@ Recommended Method
 
      and for Python 3.x, you should be able to do::
 
-      conda create --name <MYFIPYENV> --channel conda-forge python=3.6 numpy scipy matplotlib pysparse
+      conda create --name <MYFIPYENV> --channel conda-forge "python>3" numpy scipy matplotlib pysparse gmsh
```
Comment (wd15) [inline INSTALLATION.rst:91]: is `"python>3"` better than `python=3`. I guess that works when python goes to version 4 as well.

Comment (guyer) [reply INSTALLATION.rst:91]: I didn't test. I worried that might give me Python 3.0. We probably *don't* want it to start using Python 4 without warning us.

File: fipy/meshes/gmshMesh.py [hunk 2]
```diff
@@ -1387,6 +1389,11 @@ def _localOverlappingCellIDs(self):
 class Gmsh2D(Mesh2D):
     """Construct a 2D Mesh using Gmsh
 
+    If called in parallel, mesh will be partitioned based on
```
Comment (wd15) [inline fipy/meshes/gmshMesh.py:1392]: If called in parallel, the mesh will be partitioned based on the value of parallelComm.Nproc

**Review by wd15 — APPROVED** [review]

File: INSTALLATION.rst (hunk 1 above)
Comment (wd15) [reply INSTALLATION.rst:91]: I use `python=3` generally to specify the latest version that conda is using.

Comment (wd15) [reply INSTALLATION.rst:91]: with circle ci or just installing locally

Comment (guyer) [reply INSTALLATION.rst]: Thanks

File: fipy/meshes/gmshMesh.py (hunk 2 above)
Comment (guyer) [reply fipy/meshes/gmshMesh.py]: Better, thanks

**Pull request merged**